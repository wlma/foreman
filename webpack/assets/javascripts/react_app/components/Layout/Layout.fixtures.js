const mockOnClick = jest.fn();

const subItemsA = [
  {
    title: 'Aa',
    isDivider: false,
    onClick: mockOnClick,
    href: '/a',
    id: 'menu_item_aa',
  },
  {
    title: 'Cc',
    isDivider: false,
    onClick: mockOnClick,
    href: '/c',
    id: 'menu_item_cc',
  },
];
const subItemsB = [
  {
    title: 'Dd',
    isDivider: false,
    onClick: mockOnClick,
    href: '/d',
    id: 'menu_item_dd',
  },
];

const PFitems = [
  {
    title: 'Monitor',
    initialActive: true,
    iconClass: 'fa fa-tachometer',
    subItems: subItemsA,
  },
  {
    title: 'Hosts',
    initialActive: false,
    iconClass: 'fa fa-server',
    subItems: subItemsB,
  },
];
// Server Hash Data
const monitorChildren = [
  {
    type: 'item',
    name: 'Dashboard',
    title: 'Dashboard',
    exact: true,
    url: '/',
  },
  {
    type: 'item',
    name: 'Facts',
    title: 'Facts',
    url: '/fact_values',
  },
];

const hostsChildren = [
  {
    type: 'item',
    name: 'All Hosts',
    title: 'All Hosts',
    url: '/hosts/new',
  },
];

const namelessChildren = [
  {
    type: 'item',
    url: '/nameless',
  },
  {
    type: 'divider',
  },
];

const hashItemsA = [
  {
    type: 'sub_menu',
    name: 'Monitor',
    icon: 'fa fa-tachometer',
    children: monitorChildren,
  },
  {
    type: 'sub_menu',
    name: 'Hosts',
    icon: 'fa fa-server',
    children: hostsChildren,
  },
];

export const hashItemNameless = [
  {
    type: 'sub_menu',
    name: 'Empty',
    icon: 'pficon pficon-unplugged',
    children: namelessChildren,
  },
];

const logo =
  '/assets/header_logo-c9614c16f2ee399ae9cb7f36ec94b9a26bf8cf9eabaa7fe6099bf80d1f7940db.svg';
const user = {
  impersonated_by: true,
  current_user: {
    id: 4,
    login: 'admin',
    firstname: 'Admin',
    lastname: 'User',
    name: 'Admin User',
  },
  user_dropdown: [
    {
      type: 'sub_menu',
      name: 'User',
      icon: 'fa fa-user',
      children: subItemsA,
    },
  ],
};

const serverUser = {
  current_user: {
    firstname: 'G',
    lastname: 'L',
    name: 'G L',
  },
  user_dropdown: [
    {
      type: 'sub_menu',
      name: 'User',
      icon: 'fa fa-user',
      children: [
        {
          type: 'item',
          url: '/',
          name: 'My Account',
        },
        { type: 'divider' },
      ],
    },
  ],
};

export const userDropdownProps = {
  user: serverUser,
  notification_url: '/',
  isOpen: true,
};

export const fullLayoutStore = {
  layout: {
    items: [
      {
        type: 'sub_menu',
        name: 'Monitor',
        icon: 'fa fa-tachometer',
        children: [
          {
            type: 'item',
            exact: true,
            html_options: {},
            name: 'Dashboard',
            url: '/',
            title: 'Dashboard',
          },
          {
            type: 'item',
            exact: false,
            html_options: {},
            name: 'Facts',
            url: '/fact_values',
            title: 'Facts',
          },
          {
            type: 'item',
            exact: false,
            html_options: {},
            name: 'Audits',
            url: '/audits',
            title: 'Audits',
          },
          {
            type: 'divider',
            name: 'Reports',
            title: 'Reports',
          },
          {
            type: 'item',
            exact: false,
            html_options: {},
            name: 'Config Management',
            url: '/config_reports?search=eventful+%3D+true',
            title: 'Config Management',
          },
          {
            type: 'item',
            exact: false,
            html_options: {},
            name: 'Report Templates',
            url: '/templates/report_templates',
            title: 'Report Templates',
          },
        ],
        className: '',
      },
      {
        type: 'sub_menu',
        name: 'Hosts',
        icon: 'fa fa-server',
        children: [
          {
            type: 'item',
            exact: false,
            html_options: {},
            name: 'All Hosts',
            url: '/hosts',
            title: 'All Hosts',
          },
          {
            type: 'item',
            exact: false,
            html_options: {},
            name: 'Create Host',
            url: '/hosts/new',
            title: 'Create Host',
          },
          {
            type: 'divider',
            name: 'Provisioning Setup',
            title: 'Provisioning Setup',
          },
          {
            type: 'item',
            exact: false,
            html_options: {},
            name: 'Architectures',
            url: '/architectures',
            title: 'Architectures',
          },
        ],
        className: '',
      },
    ],
    isLoading: false,
    isCollapsed: false,
    currentOrganization: 'Default Organization',
    currentLocation: 'Default Location',
  },
};

export const layoutData = {
  menu: [
    {
      type: 'sub_menu',
      name: 'Monitor',
      icon: 'fa fa-tachometer',
      children: [
        {
          type: 'item',
          exact: true,
          html_options: {},
          name: 'Dashboard',
          url: '/',
        },
        {
          type: 'item',
          exact: false,
          html_options: {},
          name: 'Facts',
          url: '/fact_values',
        },
        {
          type: 'item',
          exact: false,
          html_options: {},
          name: 'Audits',
          url: '/audits',
        },
        {
          type: 'divider',
          name: 'Reports',
        },
        {
          type: 'item',
          exact: false,
          html_options: {},
          name: 'Config Management',
          url: '/config_reports?search=eventful+%3D+true',
        },
        {
          type: 'item',
          exact: false,
          html_options: {},
          name: 'Report Templates',
          url: '/templates/report_templates',
        },
      ],
    },
    {
      type: 'sub_menu',
      name: 'Hosts',
      icon: 'fa fa-server',
      children: [
        {
          type: 'item',
          exact: false,
          html_options: {},
          name: 'All Hosts',
          url: '/hosts',
        },
        {
          type: 'item',
          exact: false,
          html_options: {},
          name: 'Create Host',
          url: '/hosts/new',
        },
        {
          type: 'divider',
          name: 'Provisioning Setup',
        },
        {
          type: 'item',
          exact: false,
          html_options: {},
          name: 'Architectures',
          url: '/architectures',
        },
      ],
    },
    {
      type: 'sub_menu',
      name: 'Configure',
      icon: 'fa fa-wrench',
      children: [
        {
          type: 'item',
          exact: false,
          html_options: {},
          name: 'Host Groups',
          url: '/hostgroups',
        },
        {
          type: 'item',
          exact: false,
          html_options: {},
          name: 'Global Parameters',
          url: '/common_parameters',
        },
      ],
    },
  ],
  root: '/',
  logo,
  notification_url: '/notification_recipients',
  user,
  stop_impersonation_url: '/users/stop_impersonation',
  instance_title: 'Production',
  brand: 'foreman',
  locations: {
    current_location: 'london',
    available_locations: [
      {
        id: 2,
        title: 'london',
        href: '/locations/2-london/select',
      },
      {
        id: 3,
        title: 'norway',
        href: '/locations/3-norway/select',
      },
    ],
  },
  orgs: {
    current_org: 'org1',
    available_organizations: [
      {
        id: 1,
        title: 'org1',
        href: '/organizations/1-org1/select',
      },
      {
        id: 2,
        title: 'org2',
        href: '/organizations/2-org2/select',
      },
    ],
  },
};

export const layoutMock = {
  items: PFitems,
  data: layoutData,
  setNavigationActiveItem: jest.fn(),
};

export const noItemsMock = {
  ...layoutMock,
  items: [],
};

export const hasTaxonomiesMock = {
  ...layoutMock,
  currentLocation: 'london',
  currentOrganization: 'org1',
};
