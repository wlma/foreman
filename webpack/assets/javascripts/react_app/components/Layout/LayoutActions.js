import {
  LAYOUT_INITIALIZE,
  LAYOUT_SHOW_LOADING,
  LAYOUT_HIDE_LOADING,
  LAYOUT_EXPAND,
  LAYOUT_COLLAPSE,
} from './LayoutConstants';

export const initializeLayout = ({
  items,
  isCollapsed,
  organization,
  location,
}) => ({
  type: LAYOUT_INITIALIZE,
  payload: {
    items,
    isCollapsed,
    organization,
    location,
  },
});

export const showLoading = () => ({
  type: LAYOUT_SHOW_LOADING,
});

export const hideLoading = () => ({
  type: LAYOUT_HIDE_LOADING,
});

export const expandLayoutMenus = () => ({
  type: LAYOUT_EXPAND,
});

export const collapseLayoutMenus = () => ({
  type: LAYOUT_COLLAPSE,
});
