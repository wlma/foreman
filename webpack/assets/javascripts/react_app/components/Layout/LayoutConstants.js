export const LAYOUT_INITIALIZE = 'LAYOUT_INITIALIZE';
export const LAYOUT_SHOW_LOADING = 'LAYOUT_SHOW_LOADING';
export const LAYOUT_HIDE_LOADING = 'LAYOUT_HIDE_LOADING';
export const LAYOUT_COLLAPSE = 'LAYOUT_COLLAPSE';
export const LAYOUT_EXPAND = 'LAYOUT_EXPAND';
