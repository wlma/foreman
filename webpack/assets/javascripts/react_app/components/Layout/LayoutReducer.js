import Immutable from 'seamless-immutable';

import {
  LAYOUT_INITIALIZE,
  LAYOUT_SHOW_LOADING,
  LAYOUT_HIDE_LOADING,
  LAYOUT_EXPAND,
  LAYOUT_COLLAPSE,
} from './LayoutConstants';

const initialState = Immutable({
  items: [],
  isLoading: false,
  isCollapsed: false,
});

export default (state = initialState, action) => {
  const { payload, type } = action;

  switch (type) {
    case LAYOUT_INITIALIZE:
      return state
        .set('items', payload.items)
        .set('isCollapsed', payload.isCollapsed)
        .set('currentOrganization', payload.organization)
        .set('currentLocation', payload.location);

    case LAYOUT_SHOW_LOADING:
      return state.set('isLoading', true);

    case LAYOUT_HIDE_LOADING:
      return state.set('isLoading', false);

    case LAYOUT_EXPAND:
      return state.set('isCollapsed', false);

    case LAYOUT_COLLAPSE:
      return state.set('isCollapsed', true);

    default:
      return state;
  }
};
