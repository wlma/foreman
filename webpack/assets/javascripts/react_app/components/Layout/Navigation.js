import React, { useState, useEffect, useMemo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {
  Nav,
  NavList,
  NavItem,
  NavExpandable,
  NavItemSeparator,
} from '@patternfly/react-core';
import { getCurrentPath } from './LayoutHelper';

const titleWithIcon = (title, iconClass) => (
  <div>
    <span className={classNames(iconClass, 'nav-title-icon')} />
    <span className="nav-title">{title}</span>
  </div>
);

const Navigation = ({
  navigate,
  items,
  navigationActiveItem,
  setNavigationActiveItem,
}) => {
  const [currentPath, setCurrentPath] = useState(window.location.pathname);
  useEffect(() => {
    const handleLocationChange = () => {
      setCurrentPath(window.location.pathname);
    };
    window.addEventListener('popstate', handleLocationChange);
    return () => {
      window.removeEventListener('popstate', handleLocationChange);
    };
  }, []);

  const onMouseOver = index => {
    if (navigationActiveItem !== index) {
      setNavigationActiveItem(index);
    }
  };

  const pathFragment = path => path.split('?')[0];
  const subItemToItemMap = {};

  items.forEach(item => {
    item.subItems.forEach(subItem => {
      if (!subItem.isDivider) {
        // don't keep the query parameters for the key
        subItemToItemMap[pathFragment(subItem.href)] = item.title;
      }
    });
  });

  const groupedItems = useMemo(
    () =>
      items.map(({ subItems, ...rest }) => {
        const groups = [];
        let currIndex = 0;
        if (subItems.length) {
          if (subItems[0].isDivider) {
            groups.push({ title: subItems[0].title, groupItems: [] });
          } else {
            groups.push({ title: '', groupItems: [] });
          }
          subItems.forEach(sub => {
            if (sub.isDivider) {
              groups.push({ title: sub.title, groupItems: [] });
              currIndex++;
            } else {
              groups[currIndex].groupItems.push({
                ...sub,
                isActive: currentPath === sub.href.split('?')[0],
              });
            }
          });
        }
        return { ...rest, groups };
      }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [items.length, currentPath]
  );

  const clickAndNavigate = (_onClick, href, event) => {
    if (event.ctrlKey) return;
    event.preventDefault();
    if (_onClick && typeof _onClick === 'function') {
      _onClick();
    } else {
      navigate(href);
    }

    setCurrentPath(href);
  };
  return (
    <Nav id="foreman-nav">
      <NavList>
        {groupedItems.map(({ title, iconClass, groups, className }, index) => (
          <React.Fragment key={index}>
            <NavExpandable
              title={titleWithIcon(title, iconClass)}
              groupId="nav-expandable-group-1"
              isActive={
                subItemToItemMap[pathFragment(getCurrentPath())] === title
              }
              isExpanded={
                subItemToItemMap[pathFragment(getCurrentPath())] === title
              }
              className={className}
              onClick={() => onMouseOver(index)}
              onFocus={() => {
                onMouseOver(index);
              }}
            >
              {groups.map((group, groupIndex) =>
                groupIndex === 0 ? (
                  group.groupItems.map(
                    (
                      {
                        id,
                        title: subItemTitle,
                        className: subItemClassName,
                        href,
                        onClick,
                        isActive,
                      },
                      groupItemsIndex
                    ) => (
                      <React.Fragment key={id}>
                        <NavItem
                          className={subItemClassName}
                          id={id}
                          to={href}
                          onClick={event =>
                            clickAndNavigate(onClick, href, event)
                          }
                          isActive={isActive}
                        >
                          {subItemTitle}
                        </NavItem>
                        {groupItemsIndex !== group.groupItems.length - 1 && (
                          <NavItemSeparator />
                        )}
                      </React.Fragment>
                    )
                  )
                ) : (
                  <React.Fragment key={groupIndex}>
                    <NavItemSeparator />
                    <NavExpandable
                      title={group.title}
                      isExpanded={group.groupItems.some(
                        ({ isActive }) => isActive
                      )}
                    >
                      {group.groupItems.map(
                        (
                          {
                            id,
                            title: subItemTitle,
                            className: subItemClassName,
                            href,
                            onClick,
                            isActive,
                          },
                          groupItemsIndex
                        ) => (
                          <React.Fragment key={id}>
                            <NavItem
                              className={subItemClassName}
                              id={id}
                              to={href}
                              onClick={event =>
                                clickAndNavigate(onClick, href, event)
                              }
                              isActive={isActive}
                            >
                              {subItemTitle}
                            </NavItem>
                            {groupItemsIndex !==
                              group.groupItems.length - 1 && (
                              <NavItemSeparator />
                            )}
                          </React.Fragment>
                        )
                      )}
                    </NavExpandable>
                  </React.Fragment>
                )
              )}
            </NavExpandable>
          </React.Fragment>
        ))}
      </NavList>
    </Nav>
  );
};

Navigation.propTypes = {
  navigate: PropTypes.func.isRequired,
  items: PropTypes.array.isRequired,
  navigationActiveItem: PropTypes.number,
  setNavigationActiveItem: PropTypes.func.isRequired,
};
Navigation.defaultProps = {
  navigationActiveItem: null,
};

export default Navigation;
