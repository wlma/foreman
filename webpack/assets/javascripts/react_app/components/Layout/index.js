import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

import {
  initializeLayout,
  collapseLayoutMenus,
  expandLayoutMenus,
} from './LayoutActions';
import reducer from './LayoutReducer';
import {
  patternflyMenuItemsSelector,
  selectIsLoading,
  selectIsCollapsed,
} from './LayoutSelectors';
import { combineMenuItems } from './LayoutHelper';
import { getIsNavbarCollapsed } from './LayoutSessionStorage';
import {
  useForemanOrganization,
  useForemanLocation,
} from '../../Root/Context/ForemanContext';

import Layout from './Layout';

const ConnectedLayout = ({ children, data }) => {
  const dispatch = useDispatch();

  const currentLocation = useForemanLocation()?.title;
  const currentOrganization = useForemanOrganization()?.title;
  useEffect(() => {
    dispatch(
      initializeLayout({
        items: combineMenuItems(data),
        isCollapsed: getIsNavbarCollapsed(),
        organization: data.orgs.current_org,
        location: data.locations.current_location,
      })
    );
  }, [data, dispatch]);

  const isNavCollapsed = useSelector(state => selectIsCollapsed(state));
  useEffect(() => {
    // toggles a class in the body tag, so that the main #rails-app-content container can have the appropriate width
    if (isNavCollapsed) {
      document.body.classList.remove('pf-m-expanded');
    } else {
      document.body.classList.add('pf-m-expanded');
    }
  }, [isNavCollapsed]);
  const { push: navigate } = useHistory();
  const items = useSelector(state =>
    patternflyMenuItemsSelector(state, currentLocation, currentOrganization)
  );
  const isLoading = useSelector(state => selectIsLoading(state));
  const isCollapsed = useSelector(state => selectIsCollapsed(state));

  const [navigationActiveItem, setNavigationActiveItem] = useState(null);

  return (
    <Layout
      data={data}
      navigate={navigate}
      items={items}
      isLoading={isLoading}
      isCollapsed={isCollapsed}
      collapseLayoutMenus={() => {
        setNavigationActiveItem(null);
        dispatch(collapseLayoutMenus());
      }}
      expandLayoutMenus={() => dispatch(expandLayoutMenus())}
      navigationActiveItem={navigationActiveItem}
      setNavigationActiveItem={setNavigationActiveItem}
    >
      {children}
    </Layout>
  );
};

// export prop-types
export const { propTypes, defaultProps } = Layout;

ConnectedLayout.propTypes = {
  children: propTypes.children,
  data: propTypes.data,
};

ConnectedLayout.defaultProps = {
  children: defaultProps.children,
  data: defaultProps.data,
};

// export reducers
export const reducers = { layout: reducer };

// export connected component
export default ConnectedLayout;
