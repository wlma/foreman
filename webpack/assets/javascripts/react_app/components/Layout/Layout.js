import React from 'react';

import { Page, PageSidebar } from '@patternfly/react-core';
import { layoutPropTypes, layoutDefaultProps } from './LayoutHelper';
import Header from './components/Toolbar/Header';
import Navigation from './Navigation';
import './layout.scss';

const Layout = ({
  items,
  data,
  isLoading,
  isCollapsed,
  navigate,
  expandLayoutMenus,
  collapseLayoutMenus,
  children,
  navigationActiveItem,
  setNavigationActiveItem,
}) => {
  const onNavToggle = () => {
    if (isCollapsed) {
      expandLayoutMenus();
      document.body.classList.remove('collapsed-nav');
    } else {
      collapseLayoutMenus();
      document.body.classList.add('collapsed-nav');
    }
  };
  return (
    <>
      <Page
        mainContainerId="foreman-main-container"
        header={
          <Header data={data} onNavToggle={onNavToggle} isLoading={isLoading} />
        }
        id="foreman-page"
        sidebar={
          <PageSidebar
            isNavOpen={!isCollapsed}
            nav={
              <Navigation
                items={items}
                navigate={navigate}
                navigationActiveItem={navigationActiveItem}
                setNavigationActiveItem={setNavigationActiveItem}
              />
            }
          />
        }
      >
        {children}
      </Page>
    </>
  );
};

Layout.propTypes = layoutPropTypes;
Layout.defaultProps = layoutDefaultProps;

export default Layout;
