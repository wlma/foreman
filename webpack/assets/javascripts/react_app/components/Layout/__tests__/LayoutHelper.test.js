import {
  createInitialTaxonomy,
  combineMenuItems,
} from '../LayoutHelper';
import { layoutMock } from '../Layout.fixtures';

describe('LayoutHelper', () => {
  it('should createInitialTaxonomy', () => {
    const availableTaxonomies = [
      { id: 1, title: 'Taxonomy 1' },
      { id: 2, title: 'Taxonomy 2' },
    ];
    const [, tax2] = availableTaxonomies;
    const taxonomy = createInitialTaxonomy(tax2.title, availableTaxonomies);
    expect(taxonomy).toEqual(tax2);
  });
  it('should combineMenuItems', () => {
    const combined = combineMenuItems(layoutMock.data);
    expect(combined).toMatchSnapshot();
  });
});
