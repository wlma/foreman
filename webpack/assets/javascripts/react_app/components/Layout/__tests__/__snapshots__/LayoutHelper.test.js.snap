// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutHelper should combineMenuItems 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "exact": true,
        "html_options": Object {},
        "name": "Dashboard",
        "title": "Dashboard",
        "type": "item",
        "url": "/",
      },
      Object {
        "exact": false,
        "html_options": Object {},
        "name": "Facts",
        "title": "Facts",
        "type": "item",
        "url": "/fact_values",
      },
      Object {
        "exact": false,
        "html_options": Object {},
        "name": "Audits",
        "title": "Audits",
        "type": "item",
        "url": "/audits",
      },
      Object {
        "name": "Reports",
        "title": "Reports",
        "type": "divider",
      },
      Object {
        "exact": false,
        "html_options": Object {},
        "name": "Config Management",
        "title": "Config Management",
        "type": "item",
        "url": "/config_reports?search=eventful+%3D+true",
      },
      Object {
        "exact": false,
        "html_options": Object {},
        "name": "Report Templates",
        "title": "Report Templates",
        "type": "item",
        "url": "/templates/report_templates",
      },
    ],
    "className": "",
    "icon": "fa fa-tachometer",
    "name": "Monitor",
    "type": "sub_menu",
  },
  Object {
    "children": Array [
      Object {
        "exact": false,
        "html_options": Object {},
        "name": "All Hosts",
        "title": "All Hosts",
        "type": "item",
        "url": "/hosts",
      },
      Object {
        "exact": false,
        "html_options": Object {},
        "name": "Create Host",
        "title": "Create Host",
        "type": "item",
        "url": "/hosts/new",
      },
      Object {
        "name": "Provisioning Setup",
        "title": "Provisioning Setup",
        "type": "divider",
      },
      Object {
        "exact": false,
        "html_options": Object {},
        "name": "Architectures",
        "title": "Architectures",
        "type": "item",
        "url": "/architectures",
      },
    ],
    "className": "",
    "icon": "fa fa-server",
    "name": "Hosts",
    "type": "sub_menu",
  },
  Object {
    "children": Array [
      Object {
        "exact": false,
        "html_options": Object {},
        "name": "Host Groups",
        "title": "Host Groups",
        "type": "item",
        "url": "/hostgroups",
      },
      Object {
        "exact": false,
        "html_options": Object {},
        "name": "Global Parameters",
        "title": "Global Parameters",
        "type": "item",
        "url": "/common_parameters",
      },
    ],
    "className": "",
    "icon": "fa fa-wrench",
    "name": "Configure",
    "type": "sub_menu",
  },
  Object {
    "children": Array [
      Object {
        "name": "Any Organization",
        "onClick": [Function],
        "title": "Any Organization",
      },
      Object {
        "name": "org1",
        "onClick": [Function],
        "title": "org1",
        "type": undefined,
      },
      Object {
        "name": "org2",
        "onClick": [Function],
        "title": "org2",
        "type": undefined,
      },
    ],
    "className": "organization-menu hidden-nav-lg",
    "icon": "fa fa-building",
    "name": "Organizations",
    "type": "sub_menu",
  },
  Object {
    "children": Array [
      Object {
        "name": "Any Location",
        "onClick": [Function],
        "title": "Any Location",
      },
      Object {
        "name": "london",
        "onClick": [Function],
        "title": "london",
        "type": undefined,
      },
      Object {
        "name": "norway",
        "onClick": [Function],
        "title": "norway",
        "type": undefined,
      },
    ],
    "className": "location-menu hidden-nav-lg",
    "icon": "fa fa-globe",
    "name": "Locations",
    "type": "sub_menu",
  },
]
`;
